frontend/test/metabase/scenarios/embedding/embedding-native.cy.spec.js

```diff
@@ -79,121 +84,253 @@ const questionDetails = {
-
-    cy.contains("Affiliate").should("not.exist");
-
-    // Let's try to remove one filter
-    cy.findByText("Q2, 2018")
-      .siblings(".Icon-close")
-      .click();
-
-    // Order ID is 926 - there should be only one result after this
-    filterWidget()
-      .contains("Order ID")
-      .click();
-    cy.findByPlaceholderText("Enter an ID").type("926");
-    cy.button("Add filter").click();
-
-    cy.findByTestId("table-row").should("have.length", 1);
-
-    cy.findByText("December 29, 2018, 4:54 AM");
-    cy.findByText("CO");
-    cy.findByText("Sid Mills").should("not.exist");
-
-    cy.location("search").should("eq", "?id=926&state=KS&productId=10");
+    it("should hide filters via url", () => {
+      cy.get("@questionId").then(questionId => {
+        cy.request("PUT", `/api/card/${questionId}`, {
+          enable_embedding: true,
+          embedding_params: {
+            id: "enabled",
+            product_id: "enabled",
+            state: "enabled",
+            created_at: "enabled",
+            total: "enabled",
+          },
+        });
+
+        const payload = {
+          resource: { question: questionId },
+          params: {},
+        };
+
+        // It should be possible to both set the filter value and hide it at the same time.
+        // That's the synonymous to the locked filter.
+        visitEmbeddedPage(payload, {
+          setFilters: "id=92",
+          hideFilters: "id,product_id,state,created_at,total",
+        });
+
+        cy.findByTestId("table-row").should("have.length", 1);
+        cy.findByText("92");
+
+        filterWidget().should("not.exist");
+      });
+    });
+
+    it("should set multiple filter values via url", () => {
+      cy.get("@questionId").then(questionId => {
+        cy.request("PUT", `/api/card/${questionId}`, {
+          enable_embedding: true,
+          embedding_params: {
+            created_at: "enabled",
+            source: "enabled",
+            state: "enabled",
+            total: "enabled",
+          },
+        });
+
+        const payload = {
+          resource: { question: questionId },
+          params: {},
+        };
+
+        visitEmbeddedPage(payload, {
+          setFilters: "created_at=Q2-2019&source=Organic&state=OR",
+        });
+
+        filterWidget()
+          .should("have.length", 4)
+          .and("contain", "OR")
+          .and("contain", "Q2, 2019");
+        // Why do we use input field in one filter widget but a simple `span` in the other one?
+        cy.findByDisplayValue("Organic");
+
+        // Total's value should fall back to the default one (`0`) because we didn't set it explicitly
+        cy.get("legend")
+          .contains("Total")
+          .parent("fieldset")
+          .contains("0");
+
+        cy.contains("Emilie Goyette");
+        cy.contains("35.7");
+
+        // OTOH, we should also be able to override the default filter value by eplixitly setting it
+        visitEmbeddedPage(payload, {
```

Comment:
The beauty of the API approach and this helper is that it gives us the ability to test multiple combinations of filter values and/or hidden filters in the same test.